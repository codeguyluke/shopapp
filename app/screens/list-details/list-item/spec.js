import React from 'react'
import { shallow } from 'enzyme'
import noop from 'lodash/noop'
import ListItem from './index'

describe('ListItem', () => {
  const itemMock = {
    id: 'abc',
    name: 'Item',
    checked: true,
  }
  const component = (
    <ListItem item={itemMock} archived={false} onUpdate={noop} onToggle={noop} onDelete={noop} />
  )

  it('matches snapshot', () => {
    const wrapper = shallow(component)
    expect(wrapper).toMatchSnapshot()
  })

  it('matches snapshot when archived', () => {
    const wrapper = shallow(React.cloneElement(component, { archived: true }))
    expect(wrapper).toMatchSnapshot()
  })

  it('toggles item', () => {
    const onToggleSpy = jest.fn()
    const wrapper = shallow(React.cloneElement(component, { onToggle: onToggleSpy }))
    wrapper.find('CheckBox').simulate('press')
    expect(onToggleSpy).toHaveBeenCalledTimes(1)
    expect(onToggleSpy).toHaveBeenCalledWith({ id: 'abc' })
  })

  it("updates item's name", () => {
    const onUpdateSpy = jest.fn()
    const wrapper = shallow(React.cloneElement(component, { onUpdate: onUpdateSpy }))
    wrapper.find('TextInput').simulate('changeText', 'test name')
    expect(onUpdateSpy).toHaveBeenCalledTimes(1)
    expect(onUpdateSpy).toHaveBeenCalledWith({ id: 'abc', name: 'test name' })
  })

  it("deletes item", () => {
    const onDeleteSpy = jest.fn()
    const wrapper = shallow(React.cloneElement(component, { onDelete: onDeleteSpy }))
    wrapper.find('Button').simulate('press')
    expect(onDeleteSpy).toHaveBeenCalledTimes(1)
    expect(onDeleteSpy).toHaveBeenCalledWith({ id: 'abc' })
  })
})
