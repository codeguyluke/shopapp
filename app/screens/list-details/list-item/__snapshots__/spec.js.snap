// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem matches snapshot 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "borderBottomWidth": 1,
      "borderColor": "lightgrey",
      "flex": 1,
      "flexDirection": "row",
      "padding": 8,
    }
  }
>
  <CheckBox
    center={false}
    checked={true}
    checkedColor="green"
    checkedIcon="check-square-o"
    containerStyle={
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
        "margin": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "padding": 0,
        "width": 32,
      }
    }
    disabled={false}
    iconRight={false}
    right={false}
    size={24}
    uncheckedColor="#bfbfbf"
    uncheckedIcon="square-o"
  />
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    editable={true}
    key="abc"
    onChangeText={[Function]}
    opacity={1}
    placeholder="Shopping item"
    style={
      Object {
        "color": "midnightblue",
        "flex": 1,
        "fontFamily": "Hind-Medium",
        "fontSize": 16,
        "paddingHorizontal": 8,
        "paddingVertical": 8,
      }
    }
    underlineColorAndroid="transparent"
    value="Item"
  />
  <Button
    buttonStyle={
      Object {
        "backgroundColor": "transparent",
        "padding": 8,
        "paddingRight": 0,
      }
    }
    containerViewStyle={
      Object {
        "height": "100%",
        "marginLeft": 0,
        "marginRight": 0,
        "paddingLeft": 16,
        "paddingRight": 0,
      }
    }
    icon={
      Object {
        "color": "crimson",
        "name": "remove-circle",
        "size": 24,
      }
    }
  />
</Component>
`;

exports[`ListItem matches snapshot when archived 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "borderBottomWidth": 1,
      "borderColor": "lightgrey",
      "flex": 1,
      "flexDirection": "row",
      "padding": 8,
    }
  }
>
  <CheckBox
    center={false}
    checked={true}
    checkedColor="lightgrey"
    checkedIcon="check-square-o"
    containerStyle={
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
        "margin": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "padding": 0,
        "width": 32,
      }
    }
    disabled={true}
    iconRight={false}
    right={false}
    size={24}
    uncheckedColor="#bfbfbf"
    uncheckedIcon="square-o"
  />
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    editable={false}
    key="abc"
    onChangeText={[Function]}
    opacity={0.6}
    placeholder="Shopping item"
    style={
      Object {
        "color": "midnightblue",
        "flex": 1,
        "fontFamily": "Hind-Medium",
        "fontSize": 16,
        "paddingHorizontal": 8,
        "paddingVertical": 8,
      }
    }
    underlineColorAndroid="transparent"
    value="Item"
  />
</Component>
`;
